File: csharp/src/Training.Workshop/M8/S01BranchByAbstraction/Step1/BookingService.cs
using System.Globalization;

namespace Training.Workshop.M8.S01BranchByAbstraction.Step1;

/// <summary>Krok 1: potwierdzenie deleguje wyliczenie ceny do wydzielonej klasy LegacyTicketPricing.</summary>
public sealed class BookingService
{
    private readonly LegacyTicketPricing _pricing = new();

    public string Confirm(BookingRequest request)
    {
        double total = _pricing.Total(request);
        return request.Screening.Title + ": " + string.Join(",", request.Seats)
            + " - do zaplaty " + total.ToString("F2", CultureInfo.InvariantCulture);
    }
}

File: typescript/src/workshop/m8/s01_branchbyabstraction/step1/BookingService.ts
import type { BookingRequest } from '../BookingRequest.js';
import { LegacyTicketPricing } from './LegacyTicketPricing.js';

/** Krok 1: potwierdzenie deleguje wyliczenie ceny do wydzielonej klasy LegacyTicketPricing. */
export class BookingService {
  private readonly pricing = new LegacyTicketPricing();

  confirm(request: BookingRequest): string {
    const total = this.pricing.total(request);
    return request.screening.title + ': ' + request.seats.join(',')
      + ' - do zaplaty ' + total.toFixed(2);
  }
}
